src/logic/azext_logic/generated/custom.py

```diff
@@ -0,0 +1,176 @@
+
+
+def logic_workflow_list(cmd, client,
+                        resource_group_name=None,
+                        top=None,
+                        filter=None):
+    if resource_group_name:
+        return client.list_by_resource_group(resource_group_name=resource_group_name,
+                                             top=top,
+                                             filter=filter)
+    return client.list_by_subscription(top=top,
+                                       filter=filter)
+
+
+def logic_workflow_show(cmd, client,
+                        resource_group_name,
+                        name):
+    return client.get(resource_group_name=resource_group_name,
+                      workflow_name=name)
+
+
+def logic_workflow_create(cmd, client,
+                          resource_group_name,
+                          name,
+                          definition,
+                          location,
+                          tags=None,
+                          state=None,
+                          endpoints_configuration=None,
+                          access_control=None,
+                          integration_account=None,
+                          integration_service_environment=None):
+
+    if 'definition' not in definition:
+        raise CLIError(str(definition) +
+                       " does not contain a 'definition' key")
+
+    return client.create_or_update(resource_group_name=resource_group_name,
+                                   workflow_name=name,
+                                   location=location,
+                                   tags=tags,
+                                   state=state,
+                                   endpoints_configuration=endpoints_configuration,
+                                   access_control=definition.get(
+                                       'accessControl', access_control),
+                                   integration_account=integration_account,
+                                   integration_service_environment=integration_service_environment,
+                                   definition=definition['definition'],
+                                   parameters=definition.get('parameters', None))
+
+
+def logic_workflow_update(cmd, client,
+                          resource_group_name,
+                          name,
+                          definition,
+                          tags=None,
+                          state=None):
+
+    # check workflow exist before another update is done via a put
+    # per dicussion with the logic service team and to match powershells
+    # behavior
+    workflow = client.get(resource_group_name=resource_group_name,
+                          workflow_name=name)
+    return logic_workflow_create(cmd, client, resource_group_name, name,
+                                 definition, workflow.location,
+                                 tags if tags else workflow.tags,
+                                 state,
```

Comment:
These 2 properties have different logic. Why not use `state if state else workflow.state,`?

Reply:
Good Catch! This is a bug, fixed.